data_diff/diff_tables.py

```diff
@@ -100,17 +101,32 @@ def __iter__(self):
             self.result_list.append(i)
             yield i
 
-    def _get_stats(self) -> DiffStats:
+    def _get_stats(self, is_dbt:bool = False) -> DiffStats:
```

Comment:
Not blocking, but the logic in this method is getting very dense with these added is_dbt sections. It's pretty hard for me to follow what's going on.

With for loops nested in for loops, I wonder how this scales as diff sizes grow.

Since this is limited to the beta `--dbt` feature, I think as long as we test that diff stats are returning in \~reasonable\~ time-frames, we could tune + refactor later. This does not have test coverage currently so we should test manually with non-trivial diffs. I'd like to avoid `--dbt` hanging for any testers with medium-large diffs

Reply:
I agree, it's much more confusing now that the logic isn't duplicated across two methods. I ran it locally (in the context of `data-diff --dbt`) and it seemed pretty quick, so I don't think it's an immediate performance issue.

Reply:
But I definitely want to refactor this to make it the right balance of readable and not-redunant.